packages/terra-button/src/Button.jsx

```diff
@@ -139,7 +139,9 @@ class Button extends React.Component {
     // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
     // This will put focus on the button when clicked if it is not currently the active element.
     if (document.activeElement !== event.currentTarget) {
+      this.shouldShowFocus = false;
       event.currentTarget.focus();
+      setTimeout(() => { this.shouldShowFocus = true; });
```

Comment:
Is the timeout necessary? Occasionally unset timeouts are used has a workaround for updates in certain lifecycles, but it may not be necessary here.

Reply:
changed here b5ea39252dacda797d3d046bf3c4a1e6183b80e3